core/src/program.rs

```diff
@@ -437,7 +437,9 @@ impl<EC: EvalCache> Program<EC> {
 
     /// Load, parse, and typecheck the program and the standard library, if not already done.
     pub fn typecheck(&mut self) -> Result<(), Error> {
-        self.vm.import_resolver_mut().parse(self.main_id)?;
+        self.vm
+            .import_resolver_mut()
+            .parse(self.main_id, InputFormat::Nickel)?;
```

Comment:
Same, I think any `parse` from within `Program` methods should probably try to guess the format, to have the same behavior as `prepare`.